docs/reference/search/searching.rst

```diff
@@ -65,211 +65,111 @@ This should work the same way for Documents and custom models as well.
-  </form>
+An example page search view
+===========================
 
-The form is submitted to the URL of the ``wagtailsearch_search`` view, with the search terms variable ``q``. The view will use its own basic search results template.
-
-Let's use our own template for the results, though. First, in your project's ``settings.py``, define a path to your template:
+You can add a "Search" page to your site using a Django view. Here's an example that you can use:
 
 .. code-block:: python
 
-  WAGTAILSEARCH_RESULTS_TEMPLATE = 'mysite/search_results.html'
-
-Next, let's look at the template itself:
-
-.. code-block:: django
-
-  {% extends "mysite/base.html" %}
-  {% load wagtailcore_tags %}
-
-  {% block title %}Search{% if search_results %} Results{% endif %}{% endblock %}
-
-  {% block search_box %}
-    {% include "mysite/includes/search_box.html" with query_string=query_string only %}
-  {% endblock %}
-
-  {% block content %}
-    <h2>Search Results{% if request.GET.q %} for {{ request.GET.q }}{% endif %}</h2>
-    <ul>
-      {% for result in search_results %}
-        <li>
-          <h4><a href="{% pageurl result.specific %}">{{ result.specific }}</a></h4>
-          {% if result.specific.search_description %}
-            {{ result.specific.search_description|safe }}
-          {% endif %}
-        </li>
-      {% empty %}
-        <li>No results found</li>
-      {% endfor %}
-    </ul>
-  {% endblock %}
+    # views.py
 
-The search view provides a context with a few useful variables.
+    from django.shortcuts import render
+    from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 
-  ``query_string``
-    The terms (string) used to make the search.
+    from wagtail.wagtailcore.models import Page
+    from wagtail.wagtailsearch.models import Query
 
-  ``search_results``
-    A collection of Page objects matching the query. The ``specific`` property of ``Page`` will give the most-specific subclassed model object for the Wagtail page. For instance, if an ``Event`` model derived from the basic Wagtail ``Page`` were included in the search results, you could use ``specific`` to access the custom properties of the ``Event`` model (``result.specific.date_of_event``).
 
-  ``is_ajax``
-    Boolean. This returns Django's ``request.is_ajax()``.
+    def search(request):
+        # Search
+        search_query = request.GET.get('query', None)
+        if search_query:
+            search_results = Page.objects.live().search(search_query)
 
-  ``query``
-    A Wagtail ``Query`` object matching the terms. The ``Query`` model provides several class methods for viewing the statistics of all queries, but exposes only one property for single objects, ``query.hits``, which tracks the number of time the search string has been used over the lifetime of the site. ``Query`` also joins to the Editor's Picks functionality though ``query.editors_picks``. See :ref:`editors-picks`.
+            # Record hit
```

Comment:
does this need more explanation? e.g. "Log the query so Wagtail can suggest promoted results".